snetd/cmd/components.go

```diff
@@ -190,12 +191,12 @@ func (components *Components) GrpcInterceptor() grpc.StreamServerInterceptor {
 
 	if !components.Blockchain().Enabled() {
 		log.Info("Blockchain is disabled: no payment validation")
-		components.grpcInterceptor = handler.NoOpInterceptor
+		components.grpcInterceptor = grpc_middleware.ChainStreamServer(handler.GrpcRateLimitInterceptor(), handler.NoOpInterceptor)
 	} else {
 		log.Info("Blockchain is enabled: instantiate payment validation interceptor")
-		components.grpcInterceptor = handler.GrpcStreamInterceptor(
-			components.EscrowPaymentHandler(),
-		)
+		components.grpcInterceptor =
+			grpc_middleware.ChainStreamServer(handler.GrpcRateLimitInterceptor(), handler.GrpcStreamInterceptor(
+				components.EscrowPaymentHandler()))
```

Comment:
Minor and more about code duplication and readability: you can introduce new component ```GrpcPaymentValidationInterceptor``` and move ```if !components.Blockchain().Enabled()``` condition into component initialization procedure, then you will have ```GrpcPaymentValidationInterceptor``` component which is renamed version of old ```GrpcInterceptor```:
```golang
 func (components *Components) GrpcPaymentValidationInterceptor() grpc.StreamServerInterceptor {
 	if components.grpcPaymentValidationInterceptor != nil {
 		return components.grpcPaymentValidationInterceptor
 	}
	if !components.Blockchain().Enabled() {
		log.Info("Blockchain is disabled: no payment validation")
		components.grpcPaymentValidationInterceptor = handler.NoOpInterceptor
	} else {
		log.Info("Blockchain is enabled: instantiate payment validation interceptor")
		components.grpcPaymentValidationInterceptor = handler.GrpcPaymentValidationInterceptor(components.EscrowPaymentHandler())
        }
	return components.grpcPaymentValidationInterceptor
}
```
And new ```GrpcInterceptor``` which chains other interceptor:
```golang
 func (components *Components) GrpcInterceptor() grpc.StreamServerInterceptor {
 	if components.grpcInterceptor != nil {
 		return components.grpcInterceptor
 	}
	components.grpcInterceptor = grpc_middleware.ChainStreamServer(handler.GrpcRateLimitInterceptor(), components.GrpcPaymentValidationInterceptor())
	return components.grpcInterceptor
}
```

Reply:
Makes Sense, Fixed as a separate commit